src/libpgmoneta/walfile/wal_reader.c

```diff
@@ -209,6 +209,8 @@ pgmoneta_wal_parse_wal_file(char* path, struct server* server_info)
          size_t bytes_read = fread(buffer, 1, data_length, file);
 
          assert(bytes_read == data_length);
+
```

Comment:
This is a separate patch that @sh-soltanpour needs to approve... And, he is using `assert`'s for his code